### ISSUE TITLE
Get rid of styled-jsx

### ISSUE THREAD

**Issue by Betree** [opencollective/opencollective#3289]
According to https://github.com/opencollective/opencollective/blob/master/rfcs/002-styled-components.md, `styled-components` is the default way to write CSS in our frontend codebase, and it's mature enough to be the only one. However, there are still many places using `styled-jsx` which creates some inconsistencies and introduces [issues](https://github.com/opencollective/opencollective-frontend/blob/00688a2801eedf22d8d45c88199f5a3cf56ec72e/pages/_document.js#L36) with our setup.

We need to:
- [x] Migrate all components (see [the list](https://github.com/opencollective/opencollective/issues/3289#issuecomment-654120071) below)
- [x] Remove the code for styled-jsx in `_document`
- [x] Remove the `styled-jsx` dependency

**Comment by SudharakaP** [comment]
@Betree : I'll take this one, if no one is working on it. It will take a me probably a bit more time than usual to finish (several weeks probably) since I have some other things in life to attend to as well, if that's okay. :smile:

**Comment by znarf** [comment]
There are some components (most?) that might be deprecated soon, for these, we should not invest too much energy and maybe just convert "Styled JSX" to vanilla CSS (put all the styles in a global css).

We need to identify the components that require a true conversion to styled-components, @Betree can you provide that?

**Comment by Betree** [comment]
# Components that are going to be removed soon

- Button (#3223)
- CollectiveCard (https://github.com/opencollective/opencollective/issues/3312)
- CollectiveCategoryPicker (https://github.com/opencollective/opencollective/issues/3313)
- Comment
- CommentForm
- Comments

- HTMLEditor
- InputTypeCreditCard
- InputTypeDropzone
- InputTypeLocation
- Location
- Logo
- Member
- Membership
- MembershipsWithData
- MembersWithData
- PaymentMethodChooser
- Response
- Responses
- SectionTitle
- EditTwitterAccount
- SendMoneyToCollectiveBtn
- Sponsors


# Deprecated components (no ETA for removal)

*These require some work before they can be removed, so we should move to `styled-components` if not too complex*

- AddFundsForm
- ApplyToHostBtn (#2282)
- CollectiveCard
- InputField
- InputFieldPresets
- MarkdownEditor

# To migrate

- Body
- CollectivesWithData
- ConnectPaypal
- CreateCollectiveForm
- CreateEvent
- CreateOrganization
- EditEventForm
- EditUpdateForm
- Error
- EventSponsorCard
- GoalsCover
- Hosts
- HostsWithData
- IncognitoUserCollective
- NotFound
- NotificationBar
- PublishUpdateBtnWithData (should also use `StyledButton`)
- Updates
- UpdatesWithData
- `edit-collective/Form.js`
- `edit-collective/index.js`
- `edit-collective/Host.js`
- `edit-collective/Tiers.js`

**Comment by Betree** [comment]
Updated with component's details. PRs to migrate just part of them (or even one) are also welcome.

**Comment by SudharakaP** [comment]
> Updated with component's details. PRs to migrate just part of them (or even one) are also welcome.

@Betree : Thanks much for the update and the list of components. I will be working on this soon. :+1:

**Comment by SudharakaP** [comment]
@Betree : For `MarkOrderAsPaidBtn` and `RefundTransactionBtn` do we need to change `SmallButton` to `StyledButton`? Is that what you mean? 😄

**Comment by SudharakaP** [comment]
@Betree : Also few more things in addition to my above comment.

1) I couldn't find any styled-jsx in `CreateCollectiveForm`. So I think it's already migrated.

2) The `SmallButton.js` is not listed under [any category above](https://github.com/opencollective/opencollective/issues/3289#issuecomment-654120071). Should we migrate this as well? 

I will create a separate PR to migrate the deprecated components soon. 😄

**Comment by Betree** [comment]
> The SmallButton.js is not listed under any category above. Should we migrate this as well?

Sorry about that, `SmallButton` should not be migrated as we want to get rid of it. I've added it to the list above.

**Comment by sbinlondon** [comment]
Which ones are being updated in the host dashboard project?

* RefundTransactionBtn
* SendMoneyToCollectiveBtn

Any others?

**Comment by Betree** [comment]
I've moved `RefundTransactionBtn` and the other Orders components to "Don't migrate".

`SendMoneyToCollectiveBtn` is not part of the host dashboard, it's used by events to transfer their remaining balances to their parent.

**Comment by SudharakaP** [comment]
@Betree : I've removed `MarkdownEditor`from the list as it seems to me that it was removed. Let me know otherwise. 👍🏽

**Comment by Betree** [comment]
@SudharakaP indeed, you did good👍

**Comment by SudharakaP** [comment]
@Betree : Just to confirm no need to migrate `ApplyToHostBtn` right as it's handled in separate issue? If that's the case whats left to do is, 

- Remove the code for styled-jsx in _document
- Remove the styled-jsx dependency

both of which in my understanding needs to wait until "soon to be removed" components are removed ?

**Comment by Betree** [comment]
`ApplyToHostBtn` is indeed going to be revamped (the issue has been moved to https://github.com/opencollective/opencollective/issues/3728). But since the JSX there is quite small, it wouldn't hurt to migrate it before :-)

**Comment by SudharakaP** [comment]
> `ApplyToHostBtn` is indeed going to be revamped (the issue has been moved to #3728). But since the JSX there is quite small, it wouldn't hurt to migrate it before :-)

@Betree: Thanks. I've migrated that as well. 👍🏽

**Comment by SudharakaP** [comment]
@Betree : Feel free to ping me anytime there's more work to be done on this one. 😄

**Comment by Betree** [comment]
Thank you @SudharakaP! This issue now only depends on #3705, #3468 and #3223 (for the `Button`s) to be completed.

We can eventually migrate `AddPrepaidBudget` and `Location`, since it seems we're going to have them around a little while still.

**Comment by SudharakaP** [comment]
> Thank you @SudharakaP! This issue now only depends on #3705, #3468 and #3223 (for the `Button`s) to be completed.
> 
> We can eventually migrate `AddPrepaidBudget` and `Location`, since it seems we're going to have them around a little while still.

@Betree : I've opened two PRs migrating `AddPrepaidBudget` and `Location`. Feel free to let me know if they need any adjustments. If these are merged first I'll re-base the changes to #3223. 😄
